project.md

```diff
@@ -5,47 +5,79 @@
 
+I'm excited by this idea because it is something that I would actually use and I know other people who would benefit from it too. I came up with the idea after interviewing a CMC environmental science major, who told me about how they rely on their computer for scheduling their work. After thinking about what they said, I thought about how I interact with my computer in the same way and realized that I had a few pain points in this area. Currently, as a Google Calendar user, I have to manually input each event into the calendar which can become tedious when I have multiple similar events (study times) to enter at once. Also, Google calendar doesn't have a great full-month view which I like to use. I am excited to potentially create a solution to these pain points as I will use my project if it is built well. I find it extremely motivating to work on something that I will actually use. 
+
 ### Domain
 
 _Describe the project's domain in five words._
 
+Easy full-semester time management
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+The user would interact with the language by initially creating a calender by setting the start and end dates. Then they would have the ability to program events by specifying the day they want to edit and choosing to either add, remove or alter items that they want to schedule. The calendar would be viewed in months (that's how I view my time management) and would help students visualize and map out their semester. Programming would look like users editing, adding, and checking things off their calendars. The right way to interact with the problem domain would be to input items into the calendar as soon as one knows about them and cross them off once they are complete. Also, I want the user to be able to cross off entire days once they have passed or if there is a holiday ahead that requires no work. My DSL will hopefully be so easy to use that these tasks will take little time to do.
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When the program runs it displays a calendar with the user's inputted items. The program interacts with the user by prompting them with the option to add, move, delete, or check off an item to their calendar. Errors might occur if there are two of the same item on the same day or if the day is in the past. They will be communicated to the user with a message after they input the request. In addition to that errors might occur if things such as dates, event types, or customization requests aren't entered with the correct syntax. These errors will be communicated to the user by displaying an error message and pointing out where the program got confused.
+
```

Comment:
I think the next step might be to decide exactly what format the output calendar will be displayed (ex. excel? ical that one can put into a calendar app? word doc templates?). This will help clarify how the calendar should display after a user runs their program.

Since you mentioned wanting to go Scala for most of your project in class today, some initial libraries to help do “heavy lifting” for icals are iCal4j and biweekly in Java. Apache POI is a library for generating Microsoft documents (that I am also looking to use for my own project!), and it seems like people have build Scala wrappers for specifically excel sheets. 

[poi.scala (scala-lang.org)](https://index.scala-lang.org/folone/poi.scala)
[spoiwo (scala-lang.org)](https://index.scala-lang.org/norbert-radyk/spoiwo)